### PR TITLE
Create installer script

### PR DESCRIPTION
**Pull request by quagsirus**

### CODE REVIEW

**Comment by quagsirus** [comment]
OK, think I fixed everything now

**Comment by quagsirus** [comment]
okay, i'm finally done now, tested on 3 different machines (1 linux, 2 windows)

**Comment by AryToNeX** [comment]
I will approve those changes once we can get it tested on a macOS environment.

File: install.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+needs_ewc = False
+
+print("locating Discord install location...")
+# Get full home path
+home_dir = expanduser("~")
+try:
+    # Search for discord install within correct subdirectory
+    if platform.system() == "Linux":
+        core_dir = glob(home_dir + "/.config/discord/*/modules/discord_desktop_core")
```
Comment (AryToNeX) [inline install.py:16]: I would want this script to check if Discord is installed as a snap or flatpak package and act accordingly. As a reference, I included the correct paths in the README.

**Comment by gsbhasin123** [comment]
This script should also check for BandagedBD, Beautifuldiscord, and any other ones that are popular to ensure ease of install.
I'd be happy to add those in once this gets merged, if you want

File: install.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    exit()
+
+# Downloads default css to ED directory
+print("downloading glasscord default theme from GitHub...")
+request.urlretrieve("https://raw.githubusercontent.com/AryToNeX/Glasscord/master/glasscord_example_theme.css", ed_dir + "glasscord.css")
+
+# Downloads new css loader plugin
+print("downloading patched css loader from GitHub...")
+try:
+    rename(ed_dir + "plugins/css_loader.js", ed_dir + "plugins/css_loader.js.old")
```
Comment (AryToNeX) [inline install.py:65]: This code block is not needed anymore. Glasscord was updated to support EnhancedDiscord natively.

Comment (quagsirus) [reply install.py]: has been removed in c94f560

File: install.py (hunk 1 above)
Comment (quagsirus) [reply install.py:16]: basic support added in c94f560

**Comment by AryToNeX** [comment]
I'll merge this when we'll have macOS tested and working

**Comment by quagsirus** [comment]
> This script should also check for BandagedBD, Beautifuldiscord, and any other ones that are popular to ensure ease of install.
> I'd be happy to add those in once this gets merged, if you want
I'm happy to make the necessary changes myself if someone can provide sample index.js' with each client installed

**Comment by gsbhasin123** [comment]
For bandaged bd, the stock one:
```
require('./glasscord.js');
require('./injector');
module.exports = require('./core.asar');
```
> I'm happy to make the necessary changes myself 
Nice! ***Wonders how to help then***

File: install.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+from os.path import expanduser, isfile
+from os import rename, remove
+from urllib import request
+import json
+
+needs_ewc = False
+
+print("locating Discord install location...")
+# Get full home path
+home_dir = expanduser("~")
```
[the remark this answers is not in the document]
Comment (gsbhasin123) [reply install.py:12]: Unless, of course, we are targeting the *whole* python3 range

[the remark this answers is not in the document]
Comment (quagsirus) [reply install.py:12]: Just out of bad habit from when I started using python3, I'll update it now.

**Comment by quagsirus** [comment]
> ```
> require('./glasscord.js');
> require('./injector');
> module.exports = require('./core.asar');
> ```

Doesn't look like install folder for bBD can be detected in the same way as ED, so I'll leave this to you. I apologize for taking on work I don't have the environment to do.

**Comment by 3l0w** [comment]
i run glasscord with bBD this is the same config i have

**Comment by gsbhasin123** [comment]
> Doesn't look like install folder for bBD can be detected in the same way as ED, so I'll leave this to you. I apologize for taking on work I don't have the environment to do.

Great, and no worries!
I'll give it a shot

**Comment by AryToNeX** [comment]
BandagedBD puts its configuration folder in `.config/BetterDiscord`. I assume it's similar on Windows and Mac. Please check!

BeautifulDiscord cannot be checked easily, so don't bother about that one, unless you want to parse an entire asar file

**Comment by gsbhasin123** [comment]
> I assume it's similar on Windows and Mac. Please check!

I do know `.config/BetterDiscord` is right on linux but I don't have access to a mac or windows system, sorry :(

**Comment by JonnyBoy2000** [comment]
> I do know `.config/BetterDiscord` is right on linux but I don't have access to a mac or windows system, sorry :(

BBD's main folder is in `%appdata%/BetterDiscord` I'm pretty sure. [Screenshot](https://i.imgur.com/cCbglSz.png)

**Comment by rauenzi** [comment]
That is the location of BD's data such as settings, plugins and themes. The injection location is different.

**Comment by AryToNeX** [comment]
> That is the location of BD's data such as settings, plugins and themes. The injection location is different.

I guess they want to extract the example css theme in the correct folder, which is nice. Let them do their thing!

**Comment by rauenzi** [comment]
No problem here, just clarifying for anyone interested.

**Comment by AryToNeX** [comment]
The installer script should now download the latest asar file from the Releases section instead of relying on the staging one on the master branch.

**Comment by AryToNeX** [comment]
Closing this until further updates

**Pull request closed without merging**